### PR TITLE
pd_client: replace global config with meta storage interface

### PR DESCRIPTION
**Pull request by HuSharp**
### What is changed and how it works?

Issue Number: Ref #14484

What's Changed:

```commit-message
replace GlobalConfig to meta storage interface
```

```bash
mysql> CREATE RESOURCE GROUP IF NOT EXISTS rg1 RU_PER_SEC = 500 BURSTABLE;
Query OK, 0 rows affected (0.16 sec)

mysql> SELECT * FROM information_schema.resource_groups;
+---------+------------+----------+-----------+-------------+------------+
| NAME    | RU_PER_SEC | PRIORITY | BURSTABLE | QUERY_LIMIT | BACKGROUND |
+---------+------------+----------+-----------+-------------+------------+
| default | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
| rg1     | 500        | MEDIUM   | YES       | NULL        | NULL       |
+---------+------------+----------+-----------+-------------+------------+
2 rows in set (0.00 sec)

mysql> DROP RESOURCE GROUP IF EXISTS rg1;
Query OK, 0 rows affected (0.15 sec)

mysql> SELECT * FROM information_schema.resource_groups;
+---------+------------+----------+-----------+-------------+------------+
| NAME    | RU_PER_SEC | PRIORITY | BURSTABLE | QUERY_LIMIT | BACKGROUND |
+---------+------------+----------+-----------+-------------+------------+
| default | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
+---------+------------+----------+-----------+-------------+------------+
1 row in set (0.00 sec)
```

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
pd_client: replace global config with meta storage interface
```

### CODE REVIEW

File: components/test_pd/src/mocker/meta_storage.rs [hunk 1]
```diff
@@ -98,14 +111,28 @@ impl PdMocker for MetaStorage {
...
+                        event.set_type(mpb::EventEventType::Delete);
+                        event.set_prev_kv(convert_kv(x.pair));
+                    }
+                }
+
                 let mut resp = mpb::WatchResponse::default();
                 resp.set_events(vec![event].into());
                 sink.send((resp, Default::default())).await.unwrap();
+
+                #[cfg(feature = "failpoints")]
```
Comment (bufferflies) [inline components/test_pd/src/mocker/meta_storage.rs:127]: how about move to failpoint test dir?

**Review by bufferflies — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -199,8 +199,8 @@ procinfo = { git = "https://github.com/tikv/procinfo-rs", rev = "7693954bd1dd86e
 # When you modify TiKV cooperatively with kvproto, this will be useful to submit the PR to TiKV and the PR to
 # kvproto at the same time.
 # After the PR to kvproto is merged, remember to comment this out and run `cargo update -p kvproto`.
-# [patch.'https://github.com/pingcap/kvproto']
-# kvproto = { git = "https://github.com/your_github_id/kvproto", branch = "your_branch" }
+[patch.'https://github.com/pingcap/kvproto']
+kvproto = { git = "https://github.com/HuSharp/kvproto", branch = "support_meta_delete" }
```
Comment (bufferflies) [inline Cargo.toml:203]: Should remove this if you are ready to merge this kvproto?

**Review by YuJuncen — APPROVED** [review]

**Comment by YuJuncen** [comment]
LGTM, please merge this after the kvproto PR get merged.

File: components/test_pd/src/mocker/meta_storage.rs (hunk 1 above)
Comment (HuSharp) [reply components/test_pd/src/mocker/meta_storage.rs:127]: I haven't found a better way to move the failpoint for scenarios that need to be tested in integration test :(

**Comment by HuSharp** [comment]
@glorv friendly ping :)

**Comment by glorv** [comment]
@HuSharp  LGTM. And please reformat the code first.

**Comment by HuSharp** [comment]
wait for https://github.com/tikv/pd/pull/7096

**Pull request merged**